Lib/test/test_pathlib.py

```diff
@@ -1390,6 +1390,61 @@ def test_iterdir(self):
             expected += ['linkA', 'linkB', 'brokenLink', 'brokenLinkLoop']
         self.assertEqual(paths, { P(BASE, q) for q in expected })
 
+    def test_iterdir_recursive(self):
+        P = self.cls
+        p = P(join('dirC'))
+        it = p.iterdir(recursive=True)
+        paths = set(it)
+        expected = [('dirD', 'fileD'), ('fileC',)]
+        self.assertEqual(paths, { P(join('dirC'), *q) for q in expected })
+
+    @support.skip_unless_symlink
+    def test_iterdir_recursive_symlink(self):
+        P = self.cls
+        p = P(join('dirB'))
+        it = p.iterdir(recursive=True)
+        paths = set(it)
+        expected = [('fileB',), ('linkD', 'fileB')]
+        self.assertEqual(paths, { P(join('dirB'), *q) for q in expected })
+
+    @support.skip_unless_symlink
+    def test_iterdir_recursive_symlink_cycle(self):
+        os.mkdir(join('dirF'))
+        os.mkdir(join('dirF', 'dirG'))
+        os.mkdir(join('dirF', 'dirH'))
+        open(join('dirF', 'fileF'), 'w').close()
```

Comment:
Why not use `.touch()`? I get that this file is testing unrelated parts of `pathlib.Path`, but I'm reflexively pretty uncomfortable with `open().close()` as an idiom.

Reply:
It is how `pathlib.Path.touch` [is implemented] (https://github.com/EpicWink/cpython/blob/64e554832db355b0e1a87133151ad7d6fb953e42/Lib/pathlib.py#L1267-L1271) :D

Reply:
Maybe it is, but I would also recommend against re-implementing other existing functions as well. Consider that if a bug is fixed in the `pathlib.Path.touch` implementation, it will be fixed everywhere that function is *called*, but not everywhere that it is re-implemented.

Reply:
That makes sense: the main reason subroutines exist